src/shims/unix/fs.rs

```diff
@@ -366,14 +367,33 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 return Ok(-1);
             }
         }
+
+        let o_nofollow = this.eval_libc_i32("O_NOFOLLOW");
+        if flag & o_nofollow == o_nofollow {
+            #[cfg(unix)]
+            {
+                use std::os::unix::fs::OpenOptionsExt;
+                options.custom_flags(o_nofollow);
```

Comment:
This uses the *target* value for O_NOFOLLOW to set the flag on the host. Unless all unixes have the same value here, that's wrong -- `custom_flags` should probably use `libc::O_NOFOLLOW` or something like that.

Reply:
Ah, indeed. Fixed.